estate/models/estate_property_offer.py

```diff
@@ -0,0 +1,111 @@
+from datetime import timedelta
+
+from odoo import _, api, fields, models
+from odoo.exceptions import UserError, ValidationError
+
+
+class EstatePropertyOffer(models.Model):
+    _name = "estate.property.offer"
+    _description = "Property Offer"
+    _order = "price desc"
+    # SQL Constraints
+    _sql_constraints = [
+        (
+            "check_offer_price_positive",
+            "CHECK(price > 0)",
+            "The offer price must be strictly positive.",
+        )
+    ]
+
+    validity = fields.Integer(
+        "Valid For", default=7, help="Number of days until this offer expires"
+    )
+
+    price = fields.Float(string="Offer Price")
+    status = fields.Selection(
+        [("accepted", "Accepted"), ("refused", "Refused")], string="Status", copy=False
+    )
+    # Relational Fields
+    partner_id = fields.Many2one("res.partner", string="Buyer", required=True)
+    property_id = fields.Many2one("estate.property", string="Property", required=True)
+    property_type_id = fields.Many2one(
+        "estate.property.type",
+        string="Property Type",
+        related="property_id.property_type_id",
+        store=True,
+        readonly=True,
+    )
+    # Computed fields
+    date_deadline = fields.Datetime(
+        compute="_compute_date_deadline", inverse="_inverse_date_deadline"
+    )
+
+    @api.model
+    def create(self, vals):
```

Comment:
Very nice and robust method override but it'd be even better if you could do it in batch to increase performances.
```suggestion
    @api.model_create_multi
        def create(self, vals_list):
            for vals in vals_list:
                # Do things
            return super().create(vals_list)
```